### ISSUE TITLE
Misleading error message when gdbm_store() returns error

### ISSUE THREAD

**Issue by zostay** [jonathanstowe/p6-GDBM#1]
TL;DR: The code probably ought to check the GDBM errno and shouldn't trust documentation from GNU.

The current error message is correct as far as the documentation is concerned. That is,  the [relevant code](https://github.com/jonathanstowe/p6-GDBM/blob/master/lib/GDBM.pm#L196):

```perl6
            if $ret == -1 {
                X::GDBM::Store.new(message => "GDBM was not opened as a writer").throw;
            }
```

Does what the documentation says the truth is. However, the GDBM documentation lies (by omission).

According to the manpage on my laptop:

> If a reader calls gdbm_store, the return value will be  -1.

And according to the [online info documentation for the current release of GDBM](https://www.gnu.org.ua/software/gdbm/manual/gdbm.html#Store):

> -1
> > The item was not stored in the database because the caller was not an official writer or either key or content have a ‘NULL’ ‘dptr’ field.

As the Perl 6 type constraints ensure the `dptr` field will never be null, the error message in the exception above should be correct. However, I have a case where it is not. 

The -1 is actually just a generic "an error occurred" return value. The code actually has to check `gdbm_last_error()` to get the actual error code and `gdbm_strerror()` to turn that code into a human friendly message. In my particular case, my writes are corrupting the database somehow and the error is `GDBM_NEED_RECOVERY`, so I need to somehow insert calls to `gdbm_recover()` to my code (or figure out why in the world my database is routinely getting corrupted, but I'm not sure I really have time to dive down that rabbit hole).

I did a quick and dirty monkey-patch to fix this that I'm too ashamed of to submit directly as a PR without some revising, but I may provide one if I get it cleaned up.
